app/controllers/concerns/collection.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+# Generic methods to be used when calling resolve_collection on our models
+module Collection
+  extend ActiveSupport::Concern
+
+  include Sorting
+
+  included do
+    def resolve_collection
+      result = search(sort(policy_scope(resource)))
+      result.paginate(page: pagination_offset, per_page: pagination_limit)
+    end
+
+    private
+
+    def search(data)
+      return data if params[:search].blank?
+
+      data.search_for(params[:search])
+    end
+
+    def sort(data)
+      return data if params[:sort_by].blank?
+
+      by = params[:sort_by].each_with_object({}) do |item, obj|
+        column, direction = item.split(':')
+
+        validate_sorting(data.column_names, column, direction)
+
+        obj[column] = direction || 'asc'
+      end
+
+      data.order(by)
```

Comment:
For the first iteration of sorting this is a good start. For frontend, however, we would need to add sorting by computed columns like `ssg_version` from Profiles and `os_major_version` with `os_minor_version` from Hosts.

Reply:
Not sure how much can we deal with that on the database level :confused: at least not in a nice & simple way.